minijinja/src/vm/mod.rs

```diff
@@ -468,23 +476,27 @@ impl<'env> Vm<'env> {
                         }
                         // leave the one argument on the stack for the recursion
                         recurse_loop!(true);
-                    } else if let Some(func) = state.ctx.load(self.env, function_name) {
+                    } else if let Some(func) = state.ctx.load(self.env, name) {
                         let args = stack.slice_top(*arg_count);
                         a = ctx_ok!(func.call(state, args));
                         stack.drop_top(*arg_count);
                         stack.push(a);
                     } else {
                         bail!(Error::new(
                             ErrorKind::UnknownFunction,
-                            format!("{} is unknown", function_name),
+                            format!("{} is unknown", name),
                         ));
                     }
+
+                    state.current_call = None;
                 }
                 Instruction::CallMethod(name, arg_count) => {
+                    state.current_call = Some(name);
```

Comment:
This would need to do something like `let old_call = mem::replace(&mut state.current_call, Some(name));` since you can have things like `{{ foo.bar(blub.blah()) }}`.

Reply:
Confirmed via tests. This change also means you can get rid of the `name` argument in `Object::call_method()` if the `unwrap()` (which should always succeed) is acceptable.